packages/hash/frontend/src/lib/subgraph.ts

```diff
@@ -194,6 +194,91 @@ export const getEntity = (
   return vertex.inner;
 };
 
+/**
+ * Gets an `Entity` by its `entityId` from within the vertices of the subgraph. Throws an error if
+ * the entity couldn't be found.
+ *
+ * @param params.subgraph
+ * @param params.entityId
+ * @throws if the vertex isn't an `EntityVertex`
+ * @throws if a vertex with `entityId` doesn't exist
+ *
+ * @todo - version is required to identify a specific instance of an entity
+ *   https://app.asana.com/0/1202805690238892/1203214689883091/f
+ */
+export const mustGetEntity = (params: {
```

Comment:
Do we really need this method if we already have `getEntity`? It's a lot of boilerplate to basically do:
``` 
if (!entity) {
  throw new Error(`Vertex with entityId "${entityId}" does not exist.`);
}
```

Reply:
I'd need to duplicate this error message in 3 places, and for-see it being common for callers of `getEntity` to want the ability to just assume an entity exists in the `vertices` map (because they got the identifier out of `roots` array for instance). This also reduces the risk of someone asserting it exists using `!` rather than writing a more informative error message.

I'm therefore in favour of keeping this.

Reply:
At that point I'd kind of think we should go with one way or the other, either make `getEntity` throw an error, or make it return `undefined`. I'm not sure we should have duplicate methods with it being unclear which one you want to call.

Reply:
If I have to choose between the two implementations, I'd go for the one that throws the error for the reasons I mentioned. If you feel strongly about having only one implementation, then I'll refactor your method to stop returning `undefined`?

Reply:
Can this thread be resolved then?

Reply:
Can we leave it open for a little longer (but _**not block**_ the PR on it?)

Would like eyes from @CiaranMn about this for future shape of the 'lib'

Reply:
We should definitely make a decision on whether `getX` throws or returns something nullish if X doesn't exist. Probably throwing is better because it allows you to encode more information about why you're not getting something.

Assuming `getX` throws an error, I don't think it's a big deal if someone comes along and introduces `maybeGetX` as a helper function, to save people writing `const x = await getX().catch(() => undefined);` repeatedly

Reply:
> if someone comes along and introduces maybeGetX as a helper function

@Alfred-Mountfield how do you feel about doing it the other way around?